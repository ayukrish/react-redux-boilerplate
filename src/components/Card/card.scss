@import '../../styles/variable.scss';

.card {
  width: 300px;
  margin-bottom: 0.625rem;
  box-shadow: rgba(0, 0, 0, 0.16) 0px 2px 2px 0px,
    rgba(0, 0, 0, 0.08) 0px 0px 0px 1px;
  border-radius: 0.625rem;
  overflow: hidden;
  background: rgb(51, 51, 51);
}

.cardHeader {
  width: 100%;
  background: $black;
  max-width: 300px;
  max-height: 300px;
  position: relative;
}

.cardTitle {
  width: 100%;
  opacity: 0.8;
  position: absolute;
  bottom: 0px;
  background: $black;
  padding: 0.625rem;
  h2 {
    color: $white;
    font-size: 1.625rem;
    font-weight: 400;
  }
}

.cardBlankTitle {
  position: relative;
  background: rgb(51, 51, 51);
}

.cardInfo {
  color: $yellowtext;
  padding: 1.25rem;
}

.characterText {
  justify-content: space-between;
  padding: 0.75rem 0px 0.375rem;
  border-bottom: 1px solid rgb(68, 68, 68);

  span {
    width: 100%;
    font-size: 0.9rem;
    font-weight: 200;
    text-align: right;

    &:first-child {
      font-size: 0.7rem;
      font-weight: 400;
      color: rgb(158, 158, 158);
      text-align: left;
    }
  }
}
