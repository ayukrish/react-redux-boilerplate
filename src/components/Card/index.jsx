import React from 'react';
import PropTypes from 'prop-types';
import * as style from './card.scss';


const propTypes = {
  imgSrc: PropTypes.string,
  heading: PropTypes.string,
  dataObj: PropTypes.object
};

const defaultProps = {
  imgSrc: '',
  heading: '',
  dataObj: {}
};

const Card = ({ imgSrc, heading, dataObj }) => {
  return (
    <article className={`${style.card} flex column`}>
      <div className={style.cardHeader}>
        {imgSrc && (<img src={imgSrc} alt={heading} />)}
        <div className={`${style.cardTitle} ${!imgSrc ? style.cardBlankTitle : ''}`}>
          <h2>{heading}</h2>
        </div>
      </div>
      <div className={style.cardInfo}>
        {Object.entries(dataObj).map(([key, value]) => (
          <div
            className={`${style.characterText} flex center`}
            key={key}
          >
            <span>{key}</span>
            <span>{value}</span>
          </div>
        ))}
      </div>
    </article>
  );
};


Card.propTypes = propTypes;
Card.defaultProps = defaultProps;
export default Card;
