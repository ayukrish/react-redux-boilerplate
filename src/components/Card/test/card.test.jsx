import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

// Components
import Card from '../index';

describe('Card Test Suite', () => {
  it('renders correctly without any data', () => {
    const output = shallow(<Card />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('renders correctly without dataObj', () => {
    const output = shallow(<Card imgSrc="random Image" heading="Test" />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('renders correctly with dataObj', () => {
    const output = shallow(
      <Card
        imgSrc="random Image"
        heading="Test"
        dataObj={{
          Test: 'test'
        }}
      />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
