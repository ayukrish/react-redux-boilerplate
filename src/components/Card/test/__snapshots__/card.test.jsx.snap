// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Test Suite renders correctly with dataObj 1`] = `
<article
  className="undefined flex column"
>
  <div>
    <img
      alt="Test"
      src="random Image"
    />
    <div
      className="undefined "
    >
      <h2>
        Test
      </h2>
    </div>
  </div>
  <div>
    <div
      className="undefined flex center"
      key="Test"
    >
      <span>
        Test
      </span>
      <span>
        test
      </span>
    </div>
  </div>
</article>
`;

exports[`Card Test Suite renders correctly without any data 1`] = `
<article
  className="undefined flex column"
>
  <div>
    <div
      className="undefined undefined"
    >
      <h2 />
    </div>
  </div>
  <div />
</article>
`;

exports[`Card Test Suite renders correctly without dataObj 1`] = `
<article
  className="undefined flex column"
>
  <div>
    <img
      alt="Test"
      src="random Image"
    />
    <div
      className="undefined "
    >
      <h2>
        Test
      </h2>
    </div>
  </div>
  <div />
</article>
`;
